translator/source_code/cotrip_translator.py

```diff
@@ -0,0 +1,421 @@
+    for field in required_fields:
+        if not field:
+            return False
+            logging.warning( 'Not all required fields are present') 
+    return True
+
+
+
+# function to calculate vehicle impact
+def get_vehicle_impact(closure_type):
+
+        #Closure type value : [-'Shoulder Closure', -'Single Lane Closure', -'No Closure', -'Right Shoulder Closed', -'Alternating Traffic', '-Off-Ramp Closure',
+        # -'On-Ramp Closure', -'All Lanes Closed', -'Single Lane Traffic', -'Left Lane Closed', -'Intermittent Traffic Stops', -'Intermittent Lane Closure', 
+        # -'Right Lane Closed', -'Various Lane Closures', -'Express Lanes Closed', -'Mobile Lane Closure', -'Left Two Lanes Closed', '-Alternating Single Lane Closures',
+        # -'Left Turn Lane Closure', -'Center Lanes Closed', -'Right Turn Lane Closure', -'Intersection Closure', -'Bridge Closed'] 
+
+        #list of vehicle impact : all-lanes-closed, some-lanes-closed, all-lanes-open, alternating-one-way, unknown
+
+    all_lanes_closed = ['All Lanes Closed', 'Intersection Closure', 'Bridge Closed' ]
+    some_lanes_closed = ['Single Lane Closure', 'Left Lane Closed', 'Right Lane Closed','Various Lane Closures', 'Express Lanes Closed', 'Mobile Lane Closure', 'Left Two Lanes Closed',
+    'Left Turn Lane Closure', 'Center Lanes Closed', 'Right Turn Lane Closure', 'Alternating Single Lane Closures','Intermittent Lane Closure'  ]
+    all_lanes_open = ['No Closure', -'Intermittent Traffic Stops', 'Shoulder Closure', 'Right Shoulder Closed', 'Off-Ramp Closure', 'On-Ramp Closure',  ]
+    alternating_one_way = ['Single Lane Traffic', 'Alternating Traffic' ]
+    unknown = []
+    vehicle_impact = 'unknown'    
+    if closure_type in all_lanes_closed:
+        vehicle_impact = 'all-lanes-closed'
+    elif closure_type in some_lanes_closed:
+        vehicle_impact = 'some-lanes-closed'
+    elif closure_type in all_lanes_open:
+        vehicle_impact = 'all-lanes-open'
+    elif closure_type in alternating_one_way:
+        vehicle_impact = 'alternating-one-way'
+    return vehicle_impact
+
+
+# function to parse polyline to geometry line string
+#-104.480110,37.007645,0 -104.480103,37.008034,0 -104.480125,37.008469,0
+def parse_polyline(polylinestring):
+    if not polylinestring or type(polylinestring) != str:
+        return None
+    polyline = polylinestring.split(' ')
+    geometry=[]
+    for i in polyline:
+        coordinates = i.split(',')
+        coords=[]
+        for j in coordinates:
+            coords.append(float(j))
+        geometry.append(coords)
+    return geometry
+
+    
+
+
+
+
+
+# function to get event status
+def get_event_status(start_time_string, end_time_string):
+    start_time = dateutil.parser.parse(start_time_string)
+
+    event_status = "active"
+    if datetime.now(timezone.utc) < start_time:
+        event_status = "planned"  # if < 2 to 3 weeks make it pending instead of planned
```

Comment:
I don't see where we set this as "pending"?

Reply:
i updated the function for event_status = "pending"